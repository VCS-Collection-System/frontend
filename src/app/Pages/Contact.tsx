import React from 'react';
import { Button, Card, CardBody, CardFooter } from '@patternfly/react-core';
import { ExternalLinkAltIcon } from '@patternfly/react-icons';
import CardWithTitle from '@app/Components/CardWithTitle';
import { useHistory } from 'react-router-dom';

const Contact: React.FunctionComponent = () => {
  const history = useHistory();
  return (
    <Card isRounded>
      <CardWithTitle
        title="Help"
        info="If you need more information, use the links below."
      />
      <CardBody>
        <Button
          icon={<ExternalLinkAltIcon />}
          variant="link"
          component="a"
          href="https://redhat.service-now.com/help?id=kb_article_view&sysparm_article=KB0015661"
          target="_blank"
        >
          VCS User Guide
        </Button>
        <br />
        <Button
          icon={<ExternalLinkAltIcon />}
          variant="link"
          component="a"
          href="https://redhat.service-now.com/help?id=kb_search&spa=1&workflow_state=published&kb_knowledge_base=9f106876db5b8890ab8fe8ca4896199c"
          target="_blank"
        >
          Frequently asked questions (FAQs) and related knowledge base articles
        </Button>
        <br />
        <Button
          icon={<ExternalLinkAltIcon />}
          variant="link"
          component="a"
          href="https://redhat.service-now.com/help?id=sc_cat_item&sys_id=89e35ece1b4074102d12c880604bcb6a"
          target="_blank"
        >
          Contact People Helpdesk for process questions and systems issues
        </Button>
        <br />
        <Button
          icon={<ExternalLinkAltIcon />}
          variant="link"
          component="a"
          href="https://www.redhat.com/en/services/consulting"
          target="_blank"
        >
          Developed by the Red Hat NAPS Services Team
        </Button>

      </CardBody>
      <CardFooter>
        <Button variant="plain" onClick={() => history.push("/")}>
          Back
        </Button>
      </CardFooter>
    </Card>
  );
};

export { Contact };
