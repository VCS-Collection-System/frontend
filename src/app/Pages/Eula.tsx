import React from 'react';
import {
  Button, 
  Card,
  CardFooter,
  CardBody,
  Stack,
  Text,
} from '@patternfly/react-core';
import { CheckCircleIcon } from '@patternfly/react-icons'
import CardWithTitle from '@app/Components/CardWithTitle';

const Eula = ({login}) => {

  return (     
    <Card isRounded>
      <CardWithTitle title="Welcome" info="VCS Privacy Notice" />
      <CardBody>
        <Stack hasGutter>
          <Text>
           Please review the <a href="https://source.redhat.com/departments/legal/globallegalcompliance/compliance_folder/vcs_privacy_notice_for_red_hat_globalpdf" target="_blank"  rel="noreferrer">Covid-19 and VCS Privacy Notice available on The Source</a> for information on how Red Hat collects, uses, stores, transfers, and otherwise processes personal information about you in connection with COVID-19 and the validation check-in system (VCS) for Red Hat.
          </Text>
        </Stack>
      </CardBody>
      <CardFooter>
        <Button
          icon={<CheckCircleIcon />}
          variant="primary"
          onClick={login}>
          Accept
        </Button>
      </CardFooter>
    </Card>  
  );
};

export { Eula };
