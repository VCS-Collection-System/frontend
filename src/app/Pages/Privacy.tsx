import React from 'react';
import { Button, Card, CardBody, CardFooter } from '@patternfly/react-core';
import { ExternalLinkAltIcon } from '@patternfly/react-icons';
import CardWithTitle from '@app/Components/CardWithTitle';
import { useHistory } from 'react-router-dom';

const Privacy: React.FunctionComponent = () => {
  const history = useHistory();
  return (
    <Card isRounded>
      <CardWithTitle
        title="Privacy"
        info="Covid-19 and VCS Privacy Notice is available on The Source. Please review the link below for more information on how Red Hat collects, uses, stores, transfers, and otherwise processes personal information about you in connection with COVID-19 and the validation check-in system (VCS) for Red Hat."
      />
      <CardBody>
        <Button
          icon={<ExternalLinkAltIcon />}
          variant="link"
          component="a"
          href="https://source.redhat.com/departments/legal/globallegalcompliance/compliance_folder/vcs_privacy_notice_for_red_hat_globalpdf"
          target="_blank"
        >
          Covid-19 and VCS Privacy Notice for Red Hat
        </Button>
      </CardBody>
      <CardFooter>
        <Button variant="plain" onClick={() => history.push("/")}>
          Back
        </Button>
      </CardFooter>
    </Card>
  );
};

export { Privacy };
