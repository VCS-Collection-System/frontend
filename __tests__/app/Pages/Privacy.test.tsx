import React from 'react';
import { Privacy } from '../../../src/app/Pages/Privacy';
import { fireEvent, render, screen } from '@testing-library/react'
import reactRouterDom from "react-router-dom"

jest.mock("react-router-dom")
const mockPush = jest.fn()
reactRouterDom.useHistory = jest.fn().mockReturnValue({ push: mockPush })

describe('<Privacy />', ()=> {
  it('should be defined', () => {
    expect(Privacy).toBeDefined()
  })

  it("should match existing snapshot", () => {
    expect(render(<Privacy />).asFragment()).toMatchSnapshot()
  })

  it("should match existing snapshot", () => {
    expect(render(<Privacy />).asFragment()).toMatchSnapshot()
  })


  it("should have an external link", () => {
    render(<Privacy />)
    expect(screen.getByRole('link', {name: /Covid-19 and VCS Privacy Notice for Red Hat/i}).getAttribute('href'))
    .toEqual("https://source.redhat.com/departments/legal/globallegalcompliance/compliance_folder/vcs_privacy_notice_for_red_hat_globalpdf")
  })

  it("should return to EULA with back button", () => {
    render(<Privacy />)
    const backButton = screen.getByRole('button', {name: /Back/i})
    fireEvent.click(backButton)
    expect(mockPush).toHaveBeenCalledWith('/')
  })
})