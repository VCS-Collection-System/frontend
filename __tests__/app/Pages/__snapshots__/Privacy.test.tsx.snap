// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Privacy /> should match existing snapshot 1`] = `
<DocumentFragment>
  <article
    class="pf-c-card pf-m-rounded"
    data-ouia-component-id="OUIA-Generated-Card-1"
    data-ouia-component-type="PF4/Card"
    data-ouia-safe="true"
    id=""
  >
    <div
      class="pf-c-card__title"
    >
      <p
        class=""
        data-pf-content="true"
      >
        Privacy
      </p>
      <div
        class="pf-c-content"
      >
        <small
          class=""
          data-pf-content="true"
        >
          Covid-19 and VCS Privacy Notice is available on The Source. Please review the link below for more information on how Red Hat collects, uses, stores, transfers, and otherwise processes personal information about you in connection with COVID-19 and the validation check-in system (VCS) for Red Hat.
        </small>
      </div>
    </div>
    <div
      class="pf-c-card__body"
    >
      <a
        aria-disabled="false"
        class="pf-c-button pf-m-link"
        data-ouia-component-id="OUIA-Generated-Button-link-1"
        data-ouia-component-type="PF4/Button"
        data-ouia-safe="true"
        href="https://source.redhat.com/departments/legal/globallegalcompliance/compliance_folder/vcs_privacy_notice_for_red_hat_globalpdf"
        target="_blank"
      >
        <span
          class="pf-c-button__icon pf-m-start"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 512 512"
            width="1em"
          >
            <path
              d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
            />
          </svg>
        </span>
        Covid-19 and VCS Privacy Notice for Red Hat
      </a>
    </div>
    <div
      class="pf-c-card__footer"
    >
      <button
        aria-disabled="false"
        class="pf-c-button pf-m-plain"
        data-ouia-component-id="OUIA-Generated-Button-plain-1"
        data-ouia-component-type="PF4/Button"
        data-ouia-safe="true"
        type="button"
      >
        Back
      </button>
    </div>
  </article>
</DocumentFragment>
`;

exports[`<Privacy /> should match existing snapshot 2`] = `
<DocumentFragment>
  <article
    class="pf-c-card pf-m-rounded"
    data-ouia-component-id="OUIA-Generated-Card-2"
    data-ouia-component-type="PF4/Card"
    data-ouia-safe="true"
    id=""
  >
    <div
      class="pf-c-card__title"
    >
      <p
        class=""
        data-pf-content="true"
      >
        Privacy
      </p>
      <div
        class="pf-c-content"
      >
        <small
          class=""
          data-pf-content="true"
        >
          Covid-19 and VCS Privacy Notice is available on The Source. Please review the link below for more information on how Red Hat collects, uses, stores, transfers, and otherwise processes personal information about you in connection with COVID-19 and the validation check-in system (VCS) for Red Hat.
        </small>
      </div>
    </div>
    <div
      class="pf-c-card__body"
    >
      <a
        aria-disabled="false"
        class="pf-c-button pf-m-link"
        data-ouia-component-id="OUIA-Generated-Button-link-2"
        data-ouia-component-type="PF4/Button"
        data-ouia-safe="true"
        href="https://source.redhat.com/departments/legal/globallegalcompliance/compliance_folder/vcs_privacy_notice_for_red_hat_globalpdf"
        target="_blank"
      >
        <span
          class="pf-c-button__icon pf-m-start"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 512 512"
            width="1em"
          >
            <path
              d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
            />
          </svg>
        </span>
        Covid-19 and VCS Privacy Notice for Red Hat
      </a>
    </div>
    <div
      class="pf-c-card__footer"
    >
      <button
        aria-disabled="false"
        class="pf-c-button pf-m-plain"
        data-ouia-component-id="OUIA-Generated-Button-plain-2"
        data-ouia-component-type="PF4/Button"
        data-ouia-safe="true"
        type="button"
      >
        Back
      </button>
    </div>
  </article>
</DocumentFragment>
`;
